Makefile

```diff
@@ -12,138 +12,258 @@
+
+$(KUBECTL): ## Install kubectl
 	curl -LO https://storage.googleapis.com/kubernetes-release/release/v$(KUBERNETES_VERSION)/bin/linux/amd64/kubectl && chmod +x ./kubectl && mv kubectl /usr/local/bin/
-	make setup-kind
+
+$(TRIVY): ## Install trivy for image vulnerability scan
+	trivy -v | grep -q $(TRIVY_VERSION) || (curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v$(TRIVY_VERSION))
+
+$(BATS): ## Install bats for running the tests
+	bats --version | grep -q $(BATS_VERSION) || (apt-get -o Acquire::Retries=30 update && apt-get -o Acquire::Retries=30 install -y bats)
+
+$(ENVSUBST): ## Install envsubst for running the tests
+	envsubst -V || (apt-get -o Acquire::Retries=30 update && apt-get -o Acquire::Retries=30 install gettext-base -y)
+
+## --------------------------------------
+## Linting
+## --------------------------------------
+.PHONY: test-style
+test-style: lint lint-charts
+
+.PHONY: lint
+lint: $(GOLANGCI_LINT)
+	# Setting timeout to 5m as default is 1m
+	$(GOLANGCI_LINT) run --timeout=5m -v
+
+lint-full: $(GOLANGCI_LINT)
+	$(GOLANGCI_LINT) run -v --fast=false
+
+lint-charts: $(HELM) # Run helm lint tests
+	helm lint --strict charts/secrets-store-csi-driver
+	helm lint --strict manifest_staging/charts/secrets-store-csi-driver
+
+## --------------------------------------
+## Builds
+## --------------------------------------
+.PHONY: build
+build:
+	CGO_ENABLED=0 GOOS=linux go build -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi ./cmd/secrets-store-csi-driver
+
+.PHONY: build-windows
+build-windows:
+	CGO_ENABLED=0 GOOS=windows go build -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi ./cmd/secrets-store-csi-driver
+
+.PHONY: build-darwin
+build-darwin:
+	CGO_ENABLED=0 GOOS=darwin go build -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi ./cmd/secrets-store-csi-driver
+
+.PHONY: container
+container:
+	docker build --no-cache --build-arg LDFLAGS=$(LDFLAGS) -t $(IMAGE_TAG) -f docker/Dockerfile .
+
+.PHONY: container-linux
+container-linux: docker-buildx-builder
+	docker buildx build --no-cache --output=type=$(OUTPUT_TYPE) --platform="linux/$(ARCH)" --build-arg LDFLAGS=$(LDFLAGS) \
+ 		-t $(IMAGE_TAG)-linux-$(ARCH) -f docker/Dockerfile .
+
+.PHONY: container-windows
+container-windows: docker-buildx-builder
+	docker buildx build --no-cache --output=type=$(OUTPUT_TYPE) --platform="windows/$(ARCH)" --build-arg LDFLAGS=$(LDFLAGS) \
+		--build-arg BASEIMAGE=mcr.microsoft.com/windows/nanoserver:$(OSVERSION) \
+		--build-arg BASEIMAGE_CORE=mcr.microsoft.com/windows/servercore:$(OSVERSION) \
+ 		-t $(IMAGE_TAG)-windows-$(OSVERSION)-$(ARCH) -f docker/windows.Dockerfile .
+
+.PHONY: docker-buildx-builder
+docker-buildx-builder:
+	@if ! docker buildx ls | grep -q container-builder; then\
+		DOCKER_CLI_EXPERIMENTAL=enabled docker buildx create --name container-builder --use --driver-opt image=moby/buildkit:$(BUILDKIT_VERSION);\
```

Comment:
what is the reason for specific `BUILDKIT_VERSION`?

Reply:
The `BUILDKIT_VERSION` was initially introduced when there was a bug that caused windows image builds to fail. We can eventually remove this and rely on the default versions on the host or prow.